### ISSUE TITLE
Swedish moth names

### ISSUE THREAD

**Issue by larspett** [BiologicalRecordsCentre/ABLE#542]
Last spring we provided a list of Swedish day active moths (https://github.com/BiologicalRecordsCentre/ABLE/issues/417) that I believe was deployed but none of those seem available. 

Also, I wonder if we did provide a full list of the Swedish macro moths? If not, we can easily do that, currently we only see moth names in German in the moth part of the app. In the butterfly parts of the app, only butterflies are available despite the day active moth option being enabled. 

If you want us to provide a moth dictionary, what format and which taxonomic groups should I provide?

**Comment by larspett** [comment]
@chrisvanswaay maybe you have the format and taxonomic group info that I would need to supply Swedish common names for the full set of macromoths?

**Comment by chrisvanswaay** [comment]
No I haven't, but I think @JimBacon might help you with a start. He also arranged all the Dutch butterfly names (actually better than I did).

**Comment by larspett** [comment]
@DavidRoy @kazlauskis Here are the Swedish common micro and macro moth names in a tab separated textfile with scientific names first on each row. 
There are a few new aggregates (5) plus some mismatches between our taxonomic backbone and the ones used in the taxonomic aggregate dataset I got for NL (5). All those are listed in a separate text file. Is this the way you want the data @JimBacon ? Would be great if it could be added to the homepage (where the common names are a mixture of German and English for some reason and the app (where common names seem to be German only)
[Scientific_and_Swedish_common_names.txt](https://github.com/BiologicalRecordsCentre/ABLE/files/11029696/Scientific_and_Swedish_common_names.txt)
[Scientific_and_Swedish_common_names_additions_and_mismatches.txt](https://github.com/BiologicalRecordsCentre/ABLE/files/11029698/Scientific_and_Swedish_common_names_additions_and_mismatches.txt)

**Comment by JimBacon** [comment]
Hi,
I'm sure the file format will be fine but I do not expect to look at this for several weeks as I have a list of other jobs I am working on.
Jim.

**Comment by larspett** [comment]
The Swedish taxonomic backbone seems like it hadn't been updated against gbif etc so only one of the taxonomic mismatches were actully a mismatch. Have updated the aggregates and mismatch list here
[Scientific_and_Swedish_common_names_additions_and_mismatches_v2.txt](https://github.com/BiologicalRecordsCentre/ABLE/files/11059863/Scientific_and_Swedish_common_names_additions_and_mismatches_v2.txt)

**Comment by DavidRoy** [comment]
@kazlauskis can you review how the day-flying moths are being included in the app?
@andrewvanbreda could you add in the swedish common names when working on #464

**Comment by andrewvanbreda** [comment]
@DavidRoy Will do. 
Am just checking some things with John about the other issue first.

**Comment by larspett** [comment]
@kazlauskis is this one closing in? Am planning for some moth monitoring publicity but need to have Swedish common names in the app and website before I can do that

**Comment by kazlauskis** [comment]
Not yet, but we are actively working on the app, and we'll look into this next week.

**Comment by larspett** [comment]
Much appreciated

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod
Would you be able to get me these results please whenever you get chance.
Thanks

[SQL 1.txt](https://github.com/BiologicalRecordsCentre/ABLE/files/11157414/SQL.1.txt)
[SQL 2.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11157415/SQL.2.csv)

**Comment by BirenRathod** [comment]
@andrewvanbreda 
I attached the results.
[ABLE.zip](https://github.com/BiologicalRecordsCentre/ABLE/files/11158333/ABLE.zip)

**Comment by larspett** [comment]
Any feedback needed from my end here @andrewvanbreda? Will the names end up in the butterfly count at the same time they are deployed at the site? I'd love to promote the app more widely but all names are in German at the moment

**Comment by andrewvanbreda** [comment]
Hi @larspett

I have prepared some of the import files, but have not done the actual import yet.

The reason for the delay is the original file actually contains synonyms which are not held in the species list, so I don't think the importer will recognise them.

I am in the process of changing the import files, but I just wanted to double check with my brother (John van Breda) about this approach before I did the actual import.

So for example, one example would be "Yponomeuta irrorellus" in the import file is "Yponomeuta irrorella" in the EBMS Moths database list but many are like this.

I have not developed on or used the butterfly count app myself, however if it uses the same species list I guess the names would appear in the app.
@kazlauskis might be able to confirm that one.

Andy

**Comment by larspett** [comment]
Ok I saw those and thought I had removed them but that was probably just in the aggregates list (where this particular example is found too). Sorry about the non-standard  taxonomy used by the Swedish taxonomic backbone

**Comment by andrewvanbreda** [comment]
@larspett 
No problem. Will let you know when done.

**Comment by kazlauskis** [comment]
@andrewvanbreda is the species list update complete now? I will then fetch the new taxa to the app.

**Comment by andrewvanbreda** [comment]
@kazlauskis 
No it isn't, as have had to do a lot of preparation.
However I should be in a position to import most of it in the next day or so.
There will be about 350 items  that won't be imported initially for various reasons.

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod 
Would you be able to get me this result whenever you get a sec please. I hope it will return nothing. Cheers.

[Test if not exists SQL.txt](https://github.com/BiologicalRecordsCentre/ABLE/files/11249301/Test.if.not.exists.SQL.txt)

**Comment by BirenRathod** [comment]
@andrewvanbreda have you got my other email about using PgAdmin from your machine, so you can run this query?

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Thanks Biren, I hadn't seen the email but I have now.
I will look into fixing NPMS first, then checkout Pgadmin after that. Cheers

**Comment by andrewvanbreda** [comment]
@kazlauskis @larspett 
I have just done the import which seems to have gone as planned.
It is just under 3000 items imported, so some are deliberately left out until I can deal with those further.

For reference imported files are.
[File 1.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11265617/File.1.csv)
[File 2.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11265618/File.2.csv)
[File 3.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11265619/File.3.csv)
[File 4.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11265620/File.4.csv)

**Comment by larspett** [comment]
@andrewvanbreda is the list in production or still in DEV? German names seem to persist in the website observation lists

**Comment by kazlauskis** [comment]
@Vilius-Stankaitis can you update the species list in the app?

**Comment by JimBacon** [comment]
@Vilius-Stankaitis wait a moment while we sort something.

@andrewvanbreda your import has deleted the common names that previously existed. Are you able to restore them?

E.g. This query 
```sql
SELECT ttl.id, taxon, language, ttl.updated_on, ttl.deleted
FROM taxa_taxon_lists ttl
JOIN taxa t
	ON t.id = ttl.taxon_id
JOIN languages l
	ON l.id = t.language_id
WHERE taxon_meaning_id = 221417
```
returns

taxa_taxon_list_id | taxon | language | updated_on | deleted
---|---|---|---|---
461092 | Noctua pronuba | Latin | 2020-02-19 10:42:10.244947 | false
612966 | större bandfly | Swedish | 2023-04-18 20:17:33 | false
541306 | huismoeder | Dutch | 2023-04-18 20:17:33 | true
461093 | Hausmutter | German | 2023-04-18 20:17:33 | true

**Comment by andrewvanbreda** [comment]
@JimBacon 
I only have read access to the database, but I can query it then get biren to update.

However more concerning is this has happened, as the importer shouldn't have behaved like that.
I will raise that in Github once have confirmed that.

**Comment by johnvanbreda** [comment]
@andrewvanbreda I'm not so sure. CommonNames is an import field which groups the names together, so if you supply a commonNames value, it replaces the whole set. That's how the UI works and the importer has always been just a form of front end to the warehouse data entry UI.

**Comment by JimBacon** [comment]
It was reading your [comment](https://github.com/BiologicalRecordsCentre/ABLE/issues/225#issuecomment-620031065) here recently which made me alive to the problem, John. Worth raising an issue to warn users of this behaviour if importing common names (on any other similar field), I think.

**Comment by andrewvanbreda** [comment]
@JimBacon @johnvanbreda 
I will raise it in a bit as have a few thoughts on matter, and was involved in the project for which the existing data update functionality was added.
I have a few tests I want to do first.

**Comment by andrewvanbreda** [comment]
Hi @BirenRathod 
Are you able to run the statement below i have marked as "TEST", and if that affects 3498 rows, can you then run the other two statements. If that first statement doesn't return 3498 let me know (although it should do as have run a select statement to test it, just double-checking)

TEST
update indicia.taxa_taxon_lists
set deleted=deleted
where updated_by_id = 5553 and deleted = true
and taxon_list_id = 260 and updated_on > '2023-04-18 19:00'::timestamp;


ACTUAL STATEMENTS
update indicia.taxa_taxon_lists
set deleted=false,updated_by_id = 5553, updated_on = now()
where updated_by_id = 5553 and deleted = true
and taxon_list_id = 260 and updated_on > '2023-04-18 19:00'::timestamp;


insert into indicia.work_queue(task, entity, record_id, cost_estimate, priority, created_on)
select 'task_cache_builder_update', 'taxa_taxon_list', id, 100, 2, now()
from indicia.taxa_taxon_lists where id in (
select id
from indicia.taxa_taxon_lists
where updated_by_id = 5553 and deleted = false
and taxon_list_id = 260 and updated_on > '2023-04-18 19:00'::timestamp
);

**Comment by BirenRathod** [comment]
@andrewvanbreda It has updated exactly 3498 rows.

**Comment by andrewvanbreda** [comment]
@BirenRathod 
Brilliant thanks :)

@JimBacon
Am seeing those names restored in the Warehouse UI now.
Can also see them on the website with the correct presence ID.
Your query above also appears to be correct to me.
I don't know if you want to confirm you are happy with things before they do an app import.

Have raised an issue about the importer here btw,
https://github.com/Indicia-Team/warehouse/issues/477

Perhaps most of that issue could be argued as matter of opinion, although perhaps point 2 is quite important.

**Comment by andrewvanbreda** [comment]
Hi @larspett 
I have noticed something regarding one name on the import.
mjölkörtsspinnare is provided as the translation on both the Brahmaeidae family nam eand species Lemonia dumi.
So if this is typed into the website as the occurrence, two names appear with no visual difference between them, even though the intended meaning is different

![Screen Shot 2023-04-19 at 16 47 51](https://user-images.githubusercontent.com/2463614/233130621-1efdc9de-eeeb-4e35-b7bb-d0a7e52dd169.png)

**Comment by larspett** [comment]
Good point and something the Swedish taxonomists probably have missed. I have raised the issue with them and expect to hear back soon(?). What about an ugly hack calling the family mjölkörtsspinnare_ and fixing that once I hear back from Uppsala? So it doesn't stall the upload

**Comment by andrewvanbreda** [comment]
Hi @larspett 
I have done that, and can see it working on the website.

There are the 300 I didn't manage to upload that I will need to examine, so once you hear back from Uppsala, we can probably upload the change along with these others.

**Comment by larspett** [comment]
Did you post those 300? There was a bunch of csv:s and I wasn't sure if I should check. Looked like Uppsala used some obsolete synonyms like ella vs. ellus etc

**Comment by andrewvanbreda** [comment]
@larspett 
Yes I will post them here once I have chance to look again at the data, and write you explanations of what issues I have seen. I will try to do this as soon as I can.
But yes there will be some things for you to check. 
What you say there is indeed one of the problems.

**Comment by larspett** [comment]
By the way, we have discovered som e species complexes that ObsIdentify solves but that end up as Unknown in ButterflyCount. Can that be because of those complexes not being listed in the uploaded taxonomy? BTW still German names on Prod webpage

**Comment by andrewvanbreda** [comment]
Hi @larspett

I am not sure.
Although I have worked on many projects, I am fairly new to this particular project, so am not aware of some of the reasons why things are setup a particular way.

I can tell you the "EBMS Moths" species list does have an "Unknown" taxon item, and people can select that item during data entry as they wish.
Any species complexes would need to be present in the species list as taxon items in the same way as standard non-complex items.

If the above information does not answer your question, perhaps I could ask David to comment.

Not sure about the German, could you send me the URL address of the page where there is German that you don't think should be there.

**Comment by larspett** [comment]
It is the image analysis assigns unknown in our app but - correctly - an aggregate in ObsIdentify

**Comment by larspett** [comment]
https://butterfly-monitoring.net/elastic/all-records Swedish language set

**Comment by andrewvanbreda** [comment]
@larspett 
Ok yes I can see the German problem.
This is using the Elastic Search index.  I will raise this in Github so David can assign to the correct person.

The other issue is not an area I am familiar with. 
I will note this, and email David about it.
I think we are having a meeting about outstanding issues next week.

**Comment by larspett** [comment]
This one becomes Unknown in ButterflyCount but a species aggregate in ObsIdentify. The confidence isn't sky high here but I've been up to about 90% with the same outcome
![image](https://user-images.githubusercontent.com/19598308/233182148-661ebe1b-cf73-4625-8015-77c3d80a5176.jpeg)
![image](https://user-images.githubusercontent.com/19598308/233182163-eb042ef4-c195-4f03-a5be-d7a1f466e494.png)

**Comment by DavidRoy** [comment]
@larspett can you raise this as a separate issue.  If issues expand we never close them ;-)

**Comment by larspett** [comment]
@DavidRoy The image ID issue is now issue #549

**Comment by andrewvanbreda** [comment]
@larspett 
Likewise I have raised the language problem for you here
https://github.com/BiologicalRecordsCentre/ABLE/issues/550

**Comment by andrewvanbreda** [comment]
Hi @larspett ,

So here is a summary of what wasn't imported.

1. Rows where "NA" was provided in the Swedish column were left out.

2. Rows where the Swedish was already present in the database that didn't need importing.

3. Rows that are in the files attached (these are rows where an exact latin match wasn't found in the species list).
There are two files.
"Items with simple suggestions" with a suggested alternative name that is present in the species list and may be useable.

"Notes in detail.csv" is similar, apart from the problems aren't as simple, so the notes column contains more detail.
In some cases I have put an alternative synonym. I am not a moth expert, so these suggestions have come from an examination of the UK Master List for an alternative, and my suggestions should be checked carefully.

If you want this data to be imported, can you confirm the alternatives in both files (maybe put another column in the sheets with your notes?)

In both files, if the extra column contains nothing, it means I couldn't find an alternative.

If you think there is any new species that you need that aren't in the species list at the moment, I will need to talk to David.


[Items with simple suggestions.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11295661/Items.with.simple.suggestions.csv)

[Notes in detail.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11295662/Notes.in.detail.csv)

**Comment by larspett** [comment]
Looks like Swedish systematics isn't always that systematic. Will look into this. So the empty cells are species that don't exist in the present version of the EBMS list?

**Comment by andrewvanbreda** [comment]
Hi @larspett 
Yes the empty cell means the exact species name isn't present and I couldn't find a similar alternative so I can't suggest something.
(although it is possible there might be a suitable alternative synonym but I simply was unable to find it myself).
So I think if there are any species you definetely need that aren't present, please note them in the sheet, so they can looked at further.

If I have put notes, it means an exact match for the name wasn't found, but I was able to find a possible alternative I need you to check.

**Comment by Vilius-Stankaitis** [comment]
Hi, is this sorted? Can I update the app?

**Comment by larspett** [comment]
I have a list with corrected names where the Swedish scientific names don’t match the ebms ones 1:1. I can upload that later today if those are ok with @andrewvanbreda. Maybe @andrewvanbreda wants to check first?

**Comment by andrewvanbreda** [comment]
Hi @larspett,

If you upload it to this thread, I can look into getting it uploaded to the Warehouse (depending on David/Jim's answers below). I always review content as best I can before I do any upload so I would check first, this would purely be for spotting obvious typing mistakes.
I also need to make sure the format is ok for import.

@JimBacon @DavidRoy 
We do have a problem here (and something I was going to bring up with the Dutch upload that needs to be done).
Currently I can't use the importer without running into the problem we had last time, where adding new common names will set existing ones to deleted.
If it is done like this, we would need to use a modified version of the correction script I used after the first import.
I spoke to John, and the importer should be able to handle this, however when I have tried it, it throws errors in the situation where there is an existing common name (when tested on my own machine).
I am not sure you preference. 
I think the options are likely:
1. I use the importer, but use the similar fix I used last time to correct the database.
2. Jim to import the data directly into the database (I only have read-only warehouse access).
3. Wait for me to correct the importer and then do the import.
Even in this scenario, it is a bit tricky to do an import, as the existing common names need to be pulled into the import spreadsheet so they don't get overwritten.
So I spoke to John yesterday about this, and I will add a new column to the importer for adding a common name (that doesn't overwrite the field).
However again, this needs me to implement, and am heavily loaded at the second.
(In particular trying to making quite a few Indicia sites 8.1 compatible, which I think is fairly urgent).

**Comment by DavidRoy** [comment]
I favour option 2.  As you say, the update to PhP 8.1 is more urgent
@JimBacon ?

**Comment by JimBacon** [comment]
Okay. I have an established method for doing this.

**Comment by larspett** [comment]
[moth_list_swe_Errors_LP.xlsx](https://github.com/BiologicalRecordsCentre/ABLE/files/11354915/moth_list_swe_Errors_LP.xlsx)
Here are the two lists as separate tabs with comments included. Hope it makes sense @andrewvanbreda

**Comment by DavidRoy** [comment]
my suggestion is that we add a set of additional synonyms but keep our current list as the preferred names (aligned with the EU Moth Red List).
@larspett could you produce a simplified file with two columns: 1. current name; 2. synonym to add
@kazlauskis when creating species lists in the app, do you include both synonyms and preferred names?  If not, I think we should

**Comment by larspett** [comment]
@DavidRoy I think @andrewvanbreda wanted to have it like this to double-check. I followed his instructions & suggestions, just put it in an excel sheet. Can of course merge columns but the entries needs involving some checks at both ends I think?

**Comment by DavidRoy** [comment]
ok, I'll leave this to you and andrew.  These issues that get to 57 comments are very hard to follow.....

**Comment by andrewvanbreda** [comment]
Hi @DavidRoy @larspett 
The original idea was to change the import spreadsheet to match the database so the import would work.
However a possibility would be to add the names as new synonyms and not change the original import sheet.
Either way I think we needed Lars's feedback to make sure the alternate names I suggested are ok matches.

I think what is best is if I check what Lars has written, and then I will advise Jim on the choices he has for approach.

**Comment by andrewvanbreda** [comment]
Hi all,

OK I have tried my best to create an overview of the remaining elements of this issue so that everyone doesn't have to keep trawling going through the long thread (although situation is still a bit complicated).

1.
@larspett
Firstly
There is another file that I didn't do any imports for that is referenced above (I have re-attached it here for convenience)

[Scientific_and_Swedish_common_names_additions_and_mismatches_v2.txt](https://github.com/BiologicalRecordsCentre/ABLE/files/11365492/Scientific_and_Swedish_common_names_additions_and_mismatches_v2.txt)

Everything in this file was already present in database apart from the last line

"Scientific mismatch
Yponomeuta cagnagellus/malinellus	benveds-/apelspinnmal	cagnagellus"

There is no match for "Yponomeuta cagnagellus/malinellus" in the database, however "Yponomeuta cagnagella/malinellus" does exist. 
Lars: are you able to confirm it is OK to add the "benveds-/apelspinnmal cagnagellus" translation to that.

2.
@JimBacon @DavidRoy @larspett 
I have looked through Lars's corrections to my import, and produced an attached file I think contains everything.

[Simplified corrections.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11365459/Simplified.corrections.csv)

There are six columns as follows

1. "Import file latin (no DB match)"
The original latin provided in the import file which there is no exact match for in the EBMS Moths species list.

2. "Accepted alternative"
An alternative that I suggested that Lars has agreed to.

3. "Lars’s Request"
A name that I think Lars is suggesting should be used, however it doesn't appear to me to be present in EBMS Moths.

4. "Swedish translation"
The Swedish to import.

5. "Lars comment"
Any comments made by Lars

6. "Andrew comment"
Any comments made by myself

So essentially, it is column 2, the "Accepted alternative" that you need to import the Swedish onto.
However, this is still blank in a lot of cases, so an import won't work on those rows.
For these blank ones, you see that "Lars’s Request" column is filled. This is the name that Lars is suggesting, but it looks to me to be missing from EBMS Moths.
So in those cases you might need to work out between yourselves whether new items need adding to the EBMS Moths species list, or perhaps there are spelling issues causing there to be no match (an alternative could also be to find a synonym)

@JimBacon
I have noticed when working on other projects, that the Indicia importer needs a particular text encoding to import none-UK characters correctly (e.g. ä) without error.
I have found that "UTF-8 with bom" has been reliable in the past (obviously this might not apply to you depending on how you go about your import)

**Comment by andrewvanbreda** [comment]
@JimBacon 
Forgot to mention there is also the option that David mentioned above of importing the name from the original import file as a synonym of the existing name in the database.
So in that case I think you would be importing the name in the first column "Import file latin (no DB match)" as a synonym of the name in the second column "Accepted alternative".
I guess it just depends if you/David want those names in the species list.

**Comment by larspett** [comment]
@andrewvanbreda this looks fine, the only thing I find is that there is a traling "cagnagellus" added to the  line
"benveds-/apelspinnmal cagnagellus". 

The line should be "benveds-/apelspinnmal" and the synonym etc  should become:
"Yponomeuta cagnagella/malinellus"  <> "Yponomeuta cagnagellus/malinellus"  <>  "benveds-/apelspinnmal"

**Comment by andrewvanbreda** [comment]
Hi @larspett
I have added the Swedish, but I have not added the synonym for that yet.
David/Jim will decide whether they want the extra synonym names present, the other rows may need new entries also.

**Comment by Vilius-Stankaitis** [comment]
Hi, I have added this to the app. Let me know if any updates are required.

**Comment by larspett** [comment]
Are the translations forward only or updating present reports too? When I check on the app, previous reports are still scientific names

**Comment by Vilius-Stankaitis** [comment]
We wanted to update species data for other countries, so Swedish common moth names eg (Blåbärshakvecklare, Rönnvecklare, etc...) were included as well. 
 
https://butterflies.app.flumens.io/demo.html

**Comment by larspett** [comment]
so in demo but not in production version of app yet?

**Comment by Vilius-Stankaitis** [comment]
Production as well v1.20.0

**Comment by larspett** [comment]
OK, I notice a couple of things, My top species have the names as common names as well as scientific names but the counts are incorrect (Orthosia gothica should be 3 but are 1), the submitted moth trap surveys are still as scientific names although common names are selected. I guess this is because they were submitted as scientific names before the translation was available?

**Comment by Vilius-Stankaitis** [comment]
Mhm, I will investigate this.  Cheers

**Comment by larspett** [comment]
@Vilius-Stankaitis I try to switch between common names in Swedish and scientific names but only get Swedish ones all the time (unless I enter the scientific name manually, then it sticks). I try changing the common/scientific option in settings but nothing happens. Shouldn't it?

**Comment by Vilius-Stankaitis** [comment]
I sent you an email last week with more details (could you check the span folder). I will create a new ticket.

**Comment by larspett** [comment]
@Vilius-Stankaitis hi haven't received such an email from you directly (not in the spam folder either). Can you please resend with cc to lars.b.pettersson@gmail.com. Could have been stopped before the mailbox

**Comment by Vilius-Stankaitis** [comment]
Hi all, 

a few mistakes:

"Gammafly" is not assigned to the Swedish common name.

"Gammaeule" currently is assigned to the Swedish common names list (should be assigned to German).

Could someone fix this? 

Thank you

**Comment by JimBacon** [comment]
@Vilius-Stankaitis I've fixed the names for [_Autographa gamma_](https://warehouse1.indicia.org.uk/index.php/taxa_taxon_list/edit/459510) mentioned in your comment above, thanks.

I'm going to import another 221 Swedish moth names this afternoon that have been matched to scientific names already in our list. There are a further 88 Swedish names still to come once they have a matching scientific name in the moth list.

**Comment by JimBacon** [comment]
I've uploaded the following 232 Swedish moth names; a few more than expected as I found some synonyms among the names I thought were unmatched. I haven't added the synonyms at the moment.

[swedish-moth-names.csv](https://github.com/BiologicalRecordsCentre/ABLE/files/11531390/swedish-moth-names.csv)

Furthermore, I

- added [_Agrochola circellaris_](https://warehouse1.indicia.org.uk/index.php/taxa_taxon_list/edit/616908) with its English and Swedish names. It was missing which was surprising given it is widespread in Europe, 
- removed _Colias croceus_ from the Swedish list of missing species as it is a butterfly.
- removed `kortstreckad videguldmal`, previously given as a common name for _Phyllonorycter brevilineatella_ which is now listed as a synonym of _Phyllonorycter salicicolella_ which already has the name `vinkelbandad videguldmal` (Possibly the wrong decision - maybe both common names should be kept for this taxon.)

This leaves 74 unmatched names still to import.

**Comment by larspett** [comment]
- Phyllonorycter brevilineatellus and Phyllonorycter brevilineatella are synonyms and are called Kortstreckad videguldmal as common name
- Phyllonorycter salicicolellus and Phyllonorycter salicicolella are synoyms and are called Vinkelbandad videguldmal as common name
- These two species are distinct and not synonyms

**Comment by JimBacon** [comment]
Hi @larspett. In your spreadsheet, moth_list_swe_Errors_LP.xlsx, (in https://github.com/BiologicalRecordsCentre/ABLE/issues/542#issuecomment-1527720391) you wrote that _P.brevilineatellus_ is a synonym of _P. salicicolella_ 
I was about to conclude there was a copy and paste error in the spreadsheet but then the [Catalogue of Life](https://www.catalogueoflife.org/data/taxon/4H8GD) also states they are synonyms.
I don't know what to do!

**Comment by larspett** [comment]
ah taxonomy the joy of it :)  Georg, who compiled the list, found the synonym listing whereas I found a different (older) listing where they were not considered synonyms. All these four are apparently considered synonyms and the  common name should be Vinkelbandad videguldmal . Sorry about that

**Comment by larspett** [comment]
@JimBacon I noticed that there are no translations for species like Korscheltellus lupulinus (lupulina), are we missing common names for Hepialidae? The image recognition fails too. ObsIdentify solves the image recognition 100%

**Comment by JimBacon** [comment]
The translation for _Korscheltellus lupulina_ was one I added on Monday so I would not expect it to be present in the app yet.

Can you attach the image which is not giving the result you expect, so that I can investigate. My guess is that the photo is identified as _Korscheltellus lupulinus_ which would not be matched against our list as we have  _Korscheltellus lupulina_ without the synonym, at present.

**Comment by larspett** [comment]
Sounds likely, here it is (not cropped here) ![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/19598308/8a9b52a5-7054-4e44-ada4-0a8cdea6eaa2)

**Comment by JimBacon** [comment]
It's more depressing than I thought. The image classifier returns another synonym altogether, _Pharmacis lupulina_. 

We need to do something to align the eBMS and the NIA taxonomy.  [David's comment](https://github.com/BiologicalRecordsCentre/ABLE/issues/549#issuecomment-1516106024) suggests this is planned.

**Comment by larspett** [comment]
What is the NIA taxonomy? Couldn't easily find my way through all NIA acronyms

**Comment by JimBacon** [comment]
[Nature Identification API](https://observation.org/pages/nia-explain/)

**Comment by DavidRoy** [comment]
We are switching the updated NIA soon. We’ll also review taxonomy then

**Comment by DavidRoy** [comment]
NIA = ObsIdentify

**Comment by kazlauskis** [comment]
The app part is done for this and includes the Swedish moth names. Can we close it now?

**Comment by DavidRoy** [comment]
@larspett

**Comment by JimBacon** [comment]
I can see a comment of mine from 22 May: 'This leaves 74 unmatched names still to import'
Don't close until I have been able to check what happened to those.

There is a separate issue, https://github.com/Indicia-Team/drupal-8-modules-indicia-ai/issues/2, regarding updating the NIA classifier. This is also still to do.

**Comment by larspett** [comment]
Last time I had a look there were still numerous names in German. I can have a look now. Which ones are the unmatched names @JimBacon ?

**Comment by larspett** [comment]
This, for instance, is an attempt to view butterfly data but what appears as top entries are bumblebees and hoverfles
![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/19598308/a3ace582-82d2-4421-bb53-11c00a9cd481)

**Comment by larspett** [comment]
Some more examples:
![image](https://github.com/BiologicalRecordsCentre/ABLE/assets/19598308/4b3d26ca-592b-4a31-8e91-cf23af23e99f)

**Comment by larspett** [comment]
Looks like synonyms that are missing

- Fruktträdssommarvecklare Archips podanus
- Mattgul lavspinnare Eilema depressum
- Unknown could be Okänd

**Comment by larspett** [comment]
It seems that all these errors including those mentioned in #684 are caused  by inconsistencies between national scientific naming conventions and those used in the taxonomic backbone here. I guess there may be several more in various contries. both our issues are fairly old now and the lack of these issed being addressed complicates both the use of the app as a tool and its wider uptake
